server/src/main/java/io/unitycatalog/server/service/credential/gcp/GcpCredentialVendor.java

```diff
@@ -0,0 +1,94 @@
+package io.unitycatalog.server.service.credential.gcp;
+
+import static java.lang.String.format;
+
+import com.google.auth.oauth2.AccessToken;
+import com.google.auth.oauth2.CredentialAccessBoundary;
+import com.google.auth.oauth2.DownscopedCredentials;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.auth.oauth2.OAuth2Credentials;
+import com.google.auth.oauth2.ServiceAccountCredentials;
+import com.google.common.base.CharMatcher;
+import io.unitycatalog.server.persist.utils.ServerPropertiesUtils;
+import io.unitycatalog.server.service.credential.CredentialContext;
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import lombok.SneakyThrows;
+import org.apache.iceberg.Files;
+
+public class GcpCredentialVendor {
+
+  public static final List<String> INITIAL_SCOPES =
+      List.of("https://www.googleapis.com/auth/cloud-platform");
```

Comment:
Isn't it safer to start with `https://www.googleapis.com/auth/devstorage.read_write` as the initial scope (from the [cloud storage oauth scopes page](https://cloud.google.com/storage/docs/oauth-scopes))

Reply:
Since we are always downscoping, it doesn't really matter. At some point in the future, it's theoretically possible the server may need to do something that requires more than just read/write. Also, I don't remember if this scope is actually required when using a service account. (btw, this other oss catalog does implemented this the same way)